dev-client/src/components/modals/ActionsModal.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {
+  Modal,
+  ModalHandle,
+  ModalProps,
+} from 'terraso-mobile-client/components/modals/Modal';
+import {forwardRef, useImperativeHandle, useMemo, useRef} from 'react';
+import {
+  Box,
+  HStack,
+  Heading,
+} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {Button} from 'native-base';
+
+type ActionButtonProps = {
+  onPress: () => void;
+  variant: keyof typeof ACTION_BUTTON_VARIANTS;
+};
+
+const ACTION_BUTTON_DEFAULTS = {
+  px: '24px',
+  py: '10px',
+  _text: {
+    fontWeight: 400,
+    fontSize: '14px',
+  },
+} as const satisfies React.ComponentProps<typeof Button>;
+
+const ACTION_BUTTON_VARIANTS = {
+  default: {
+    _text: {
+      ...ACTION_BUTTON_DEFAULTS._text,
+      color: 'primary.contrast',
+    },
+  },
+  subtle: {
+    backgroundColor: 'grey.200',
+    borderWidth: '1px',
+    borderColor: 'm3.sys.light.outline',
+    _text: {
+      ...ACTION_BUTTON_DEFAULTS._text,
+      color: 'text.primary',
+    },
+  },
+  warning: {
+    backgroundColor: 'error.main',
+    _text: {
+      ...ACTION_BUTTON_DEFAULTS._text,
+      color: 'error.contrast',
+    },
+  },
+} as const satisfies Record<string, React.ComponentProps<typeof Button>>;
+
+export const ActionButton = ({
+  variant,
+  onPress,
+  children,
+}: React.PropsWithChildren<ActionButtonProps>) => {
+  return (
+    <Button
+      {...ACTION_BUTTON_DEFAULTS}
+      {...ACTION_BUTTON_VARIANTS[variant]}
+      onPress={onPress}>
+      {children}
+    </Button>
+  );
+};
+
+export type ActionsModalProps = Pick<ModalProps, 'trigger'> & {
+  title: string;
+  actions: React.ReactNode;
+};
+/**
+ * Modal presented to a user when asked to confirm a decision
+ */
+export const ActionsModal = forwardRef<
```

Comment:
question: should this be `ActionModal`? Other uses are signular.

Reply:
suggestion: add file-level comment explaining how ActionModal, ConfirmModal and Modal differ.

Reply:
re: `Action` vs `Actions`, i chose `ActionsModal` as it's a modal which can contain many actions, whereas `ActionButton` is a button to describe a particular action. i updated the comments as you asked to make it clearer